jcli/src/jcli_app/certificate/new_vote_cast.rs

```diff
@@ -3,42 +3,94 @@ use chain_impl_mockchain::{
     certificate::{Certificate, VoteCast, VotePlanId},
     vote::{Choice, Payload},
 };
+use rand_chacha::rand_core::SeedableRng;
 use std::path::PathBuf;
 use structopt::StructOpt;
 
-/// create a vote cast certificate
 #[derive(StructOpt)]
-pub struct VoteCastCmd {
+pub struct PublicVoteCast {
+    /// the vote plan identified on the blockchain
+    vote_plan_id: VotePlanId,
+
+    /// the number of proposal in the vote plan you vote for
+    proposal_index: u8,
+
+    /// the number of choice within the proposal you vote for
+    choice: u8,
+
+    /// write the output to the given file or print it to the standard output if not defined
+    #[structopt(long = "output")]
+    output: Option<PathBuf>,
+}
+
+#[derive(StructOpt)]
+pub struct PrivateVoteCast {
     /// the vote plan identified on the blockchain
-    pub vote_plan_id: VotePlanId,
+    vote_plan_id: VotePlanId,
 
     /// the number of proposal in the vote plan you vote for
-    pub proposal_index: u8,
+    proposal_index: u8,
+
+    /// size of voting options
+    options: usize,
 
     /// the number of choice within the proposal you vote for
-    pub choice: u8,
+    choice: u8,
 
-    /// should be used for vote plans with public votes
-    #[structopt(long = "public")]
-    pub public: bool,
+    /// key to encrypt the vote with
+    encrypting_key: String,
 
     /// write the output to the given file or print it to the standard output if not defined
     #[structopt(long = "output")]
-    pub output: Option<PathBuf>,
+    output: Option<PathBuf>,
 }
 
-impl VoteCastCmd {
+/// create a vote cast certificate
+#[derive(StructOpt)]
+pub enum VoteCastCmd {
+    Public(PublicVoteCast),
+    Private(PrivateVoteCast),
+}
+
+impl PublicVoteCast {
     pub fn exec(self) -> Result<(), Error> {
-        let payload = if self.public {
-            Payload::Public {
-                choice: Choice::new(self.choice),
-            }
-        } else {
-            unimplemented!("private votes are not supported yet");
+        let payload = Payload::Public {
+            choice: Choice::new(self.choice),
         };
 
         let vote_cast = VoteCast::new(self.vote_plan_id, self.proposal_index, payload);
         let cert = Certificate::VoteCast(vote_cast);
         write_cert(self.output.as_deref(), cert.into())
     }
 }
+
+impl PrivateVoteCast {
+    pub fn exec(self) -> Result<(), Error> {
+        let mut rng = rand_chacha::ChaChaRng::from_entropy();
+
+        let key_bin = base64::decode(self.encrypting_key).map_err(Error::Base64)?;
```

Comment:
Is there a convention on which format these keys should be encoded with?
We use bech32 for stake keys, maybe the voting keys should follow suit?

Reply:
Not sure about compatibilities nor conventions though. I'll take a look.

Reply:
base64 should be good for the command line on Unix, but `+` and `/` have special meaning on Windows.
Do we already use it anywhere else?

Reply:
If in doubt, use hex.

Reply:
We can use the urlsafe mode for that I think. But if we are using the other format (bech32) in other keys lets try to keep it simple and use that format for all of them.